ext/src/ruby_api/convert.rs

```diff
@@ -122,22 +127,42 @@ impl ToExtern for Value {
 }
 
 pub trait ToSym {
-    fn to_sym(self) -> Symbol;
+    fn as_sym(&self) -> Result<Symbol, Error>;
```

Comment:
`ValType` is no longer `Copy`, which is the main motivation for this change. I also suspect that once APIs for typed function reference and GC are introduced it's likely that we will have to rework the relationship between `ValType` and its Ruby representation.

Reply:
What's your rationale for the naming change?

`#to_sym` is the normal name for converting to a Symbol, so I think the name could stay the same even if it can now raise.

Reply:
The name change was mostly to reflect `owned` -> `borrow` in the signature, but then I realized that I misinterpreted this table https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reverted!